docs/src/your-own-sender.md

```diff
@@ -0,0 +1,197 @@
+---
+short-description: Brief overview of a custom sender and operation.
+...
+
+# Writing your own sender and operation
+
+While libasync provides a veriaty of existing types, one may wish to write their own senders and operations. The most common reason for writing a custom sender and operation is wrapping around an existing callback-based API while avoiding extra costs induced by allocating coroutine frames. The following section goes into detail on how to implement them by implementing a simple wrapper for libuv's `uv_write`.
+
+## End effect
+
+The sender and operation we'll implement here will allow us to do the following:
+```cpp
+auto result = co_await write(my_handle, my_bufs, n_my_bufs);
+if (result < 0)
+	/* report error */
+```
+
+## The implementation
+
+### The sender
+
+As explained in [](sender-receiver.md), the sender is an object that stores the neceessary state to start an operation. Let's start off by looking at the prototype for `uv_write`:
+```cpp
+int uv_write(uv_write_t *req, uv_stream_t *handle, const uv_buf_t bufs[], unsigned int nbufs, uv_write_cb cb);
+```
+
+The function takes some object that stores the state (not to be confused with our operation object), a handle to the stream, buffers to write and a callback. The callback is also given a status code, which we will propagate back to the user. The state and callback will be handled by our operation, which leaves only the handle, buffers and status code.
+
+Let's start by writing a simple class:
+```cpp
+struct [[nodiscard]] write_sender {
+	using value_type = int; // Status code
+
+	uv_stream_t *handle;
+	const uv_buf_t *bufs;
+	size_t nbufs;
+};
+```
+
+As can be seen, the sender class is quite simple. The `nodiscard` attribute only helps catch errors caused by accidentally ignoring the sender without awaiting it and can be omitted.
+
+Next we add a simple function used to construct the sender:
+```cpp
+write_sender write(uv_stream_t *handle, const uv_buf_t *bufs, size_t nbufs) {
+	return write_sender{handle, bufs, nbufs};
+}
+```
+
+In addition to that, we need the `connect` overload:
+```cpp
+template <typename Receiver>
+write_operation<Receiver> connect(write_sender s, Receiver r) {
+	return {s, std::move(r)};
+}
+```
+
+`connect` simply constructs an operation from the sender and receiver.
```

Comment:
link to `connect`s documentation?

Reply:
`connect` doesn't have any concrete documentation as it's an user-defined function.